### PR TITLE
Look for binaryen.js/wasm.js in install directory as well as source dir

### LINKED ISSUE #6406: wasm.js/binaryen.js assume binaryen source directory
These .js files currently live in bin/ directory of the binaryen source, but when installed they live in shared/binaryen.

I propose that we allow BINARYEN_ROOT to point to the install root, *or* the source root.  Unfortunately it can't point to the build root as these files never exist in the build root since they are copied directly from the source directory to the install directory.

### PR DESCRIPTION
**Pull request by sbc100**
Fixes #6406

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -2396,6 +2396,17 @@ def get_binaryen_bin():
     Building.get_binaryen()
     return os.path.join(Settings.BINARYEN_ROOT, 'bin')
 
+  @staticmethod
+  def get_binaryen_lib():
+    Building.get_binaryen()
+    paths = (os.path.join(Settings.BINARYEN_ROOT, 'bin'),
+             os.path.join(Settings.BINARYEN_ROOT, 'share', 'binaryen'))
```
Comment (kripken) [inline tools/shared.py]: i don't follow, what would the `share/binaryen` use case be?

i was expecting to see something like a check if `bin/` and `lib/` exist, or if we are in the `bin` dir already and need to go one up, something like that, but I guess I misunderstood?

**Comment by sbc100** [comment]
Currently BINARYEN_ROOT is the source tree then wasm.js lives in `bin`.    If you install binaryen via `make install` or `ninja install` these js files live in `share/binaryen`. 

Since we were only looking in `bin` currently that mean we don't support the installed version binaryen.

**Comment by sbc100** [comment]
I have separate change that puts these files into `lib` and then installs them into `lib/binaryen`, but that is less urgent and purely cosmetic: https://github.com/WebAssembly/binaryen/compare/move_js_libs?expand=1

**Comment by kripken** [comment]
I see, thanks. lgtm, but please add a comment in the new `get_binaryen_lib()` method - it seems surprising to see it look in `bin/` given the name is `lib`, would be good to explain that.

**Pull request merged**